### DISCUSSION TITLE
How performance optimizer works

### DISCUSSION THREAD

**Discussion by leopseft** [YoCodingMonster/OpenFreezeCenter#68] [Q&A]
Hello my friend, could you please give us some details about what does the optimizer controls? Does it controls the fans and the cpu performance?

My problem is that when i compile i see only two of my cpus use the 100% of their power while others do nothing. I have set through pkgbuild config the usage of 4 core (of the 8) but as I said only the 2 of them do the work and the temp goes up to 90 Celsius degrees, while the fans doesn't work so fast as the expected.

So do you believe it has to do with the selected optimizer from the windows msi center? I ve choosen the balanced.

Will you program "fix" those issues when you will add support for the MSI Center for the msi modern 15 models?

Thank you!!

**Comment by YoCodingMonster** [comment]
i am preety new to git, i am working on it, since its final year of my college there has been a lot of work and i am grinding through to it all. Working is slow right now, but will pickup pace!, thanks for supporting and being helpful!!!!


but the script doesn't use any of the optimizations on CPU/GPU or any sort of performance boosting options..
while my testing i can say that there is no performance issue with the script, but your problem is not solved by that i guess XD

if you can tell me your config i can help you resolve this

**Reply by leopseft** [reply]
So what does the script controls? Only the fan speed?

Which config do you need?

**Reply by YoCodingMonster** [reply]
Currently it only controls fan speed....

You can send me the details of the problem and what system specifications you have....
Any diagnostics you would have done...or any suspitions you have on problem

**Reply by leopseft** [reply]
I don't have any specific configurations on the laptop. I'm using the default Arch Kernel, no tlp or lmt or powetop for power management, neither auto-cpufreq. Everything is in stock condition. The turbo boost enabled.

Maybe this is how turbo boost works on the cpu. Maybe the two threads work at the maximum, while compiling (??)

I'm using the default intel_pstate powersave governor and job scheduler is set to none since i use nvme ssd.

Where can I find the cpu fan controls, so you can check if they seem alright, cause i ve no idea how the fan works!! Do you find it normally to reach 90 degrees and fans not working at the maximum?
